vpr/src/base/read_blif.cpp

```diff
@@ -617,6 +618,80 @@ struct BlifAllocCallback : public blifparse::Callback {
+        std::unordered_map<AtomNetId,AtomNetId>& sink_to_source, AtomNetId source)
+        {
+            for (auto tsink : source_to_sinks[source]){
+                //recurse down the tree 
+                merge_conn_trace(source_to_sinks,sink_to_source,tsink); 
+
+                if (getEchoEnabled()) {
+                    FILE* fp;
+                    fp = vtr::fopen(getEchoFileName(E_ECHO_NETLIST_CONN_MERGE_TRACE), "a");
+                    fprintf(fp,"Merging %s \n====>\n %s\n",curr_model().net_name(source).c_str(),curr_model().net_name(tsink).c_str()); 
+                    fclose(fp);
+                }
+                //do the actual merging of sink --> source 
+                curr_model().merge_nets(source,tsink); 
+
+            }
+        }
+
+    /** @brief Merges nets specified with .conn statements
+     *
+     * Supoorts chaining of .conn statement e.g: 
+     *      .conn A B
+     *                  <===> .conn A C 
+     *      .conn B C 
+     * the dependency tree created by such chaining of .conn statements
+     * is traversed recursively to ensure the correct merging order of 
+     * sink and driver nets of every statement. Recursion is started at 
+     * root nodes (driver nets that are not sinks in any .conn statement) 
+     * and is terminated at leaf-nodes (sink nets that are not sources 
+     * in any .conn statements). 
+     */
+    void merge_conn_tree() {
+        std::unordered_map<AtomNetId,std::vector<AtomNetId>> source_to_sinks; 
+        std::unordered_map<AtomNetId,AtomNetId> sink_to_source;
+        std::vector<std::pair<AtomNetId,AtomNetId>> new_conn_list; 
+        std::vector<AtomNetId> head_sources; 
+
+        //find root nodes (source nets)
+        for (auto net_pair : curr_nets_to_merge_) {
+            source_to_sinks[net_pair.first].push_back(net_pair.second);
+            sink_to_source[net_pair.second]=net_pair.first; 
+        } 
+
+        for (auto net_pair : curr_nets_to_merge_) {
+            if (sink_to_source.find(net_pair.first) == sink_to_source.end()) {
+                head_sources.push_back(net_pair.first); 
+            }
+        } 
+
+        for (AtomNetId hsource : head_sources) {
+            merge_conn_trace(source_to_sinks,sink_to_source,hsource); 
+        }
+    } 
+    
+    void dump_conn_netlist(char* filename) {
+        FILE* fp;
+        fp = vtr::fopen(filename, "a");
+
+        fprintf(fp, "--------------------------------------------------------------\n");
+        fprintf(fp, "----.Conn Specified EBLIF nets for block: %s\n",curr_model().netlist_name().c_str());
+        fprintf(fp, "--------------------------------------------------------------\n");
+        fprintf(fp, "\n");
+
+        for (auto net_pair: curr_nets_to_merge_) {
+            fprintf(fp, "%s =====> %s\n" , curr_model().net_name(net_pair.first).c_str(),curr_model().net_name(net_pair.second).c_str());
```

Comment:
This line is rather long, consider breaking it up into multiple lines. For example, something like:
```suggestion
            fprintf(fp,
                    "%s =====> %s\n" ,
                    curr_model().net_name(net_pair.first).c_str(),
                    curr_model().net_name(net_pair.second).c_str());
```